perf(codegen): unroll loop in `SourcemapBuilder::update_generated_line_and_column`

sapphi-red opened the pull request:
Unrolled the loop in `SourcemapBuilder::update_generated_line_and_column`. This improved the perf on my machine by 3 - 8% (`just benchmark-one codegen`).

I tried a closure with `#[inilne(always)]` but that didn't improve perf. So the code uses a macro instead.

Boshen commented:
### Merge activity

* **Sep 22, 5:07 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
perf(codegen): unroll loop in `SourcemapBuilder::update_generated_line_and_column` (#13903)

Unrolled the loop in `SourcemapBuilder::update_generated_line_and_column`. This improved the perf on my machine by 3 - 8% (`just benchmark-one codegen`).

I tried a closure with `#[inilne(always)]` but that didn't improve perf. So the code uses a macro instead.
